dummyfs/dummyfs.c

```diff
@@ -531,419 +563,303 @@ int dummyfs_readdir(oid_t *dir, offs_t offs, struct dirent *dent, unsigned int s
+	dummyfs_object_t *o;
+	int ret = EOK;
 
-		/* Try to mount fs as root */
-		if (portRegister(dummyfs_common.port, "/", &root) < 0) {
-			LOG("can't mount as rootfs\n");
-			return -1;
-		}
+	o = object_get(dummyfs, oid->id);
 
-#ifdef TARGET_IMX6ULL
-		portDestroy(reserved);
-#endif
-	}
-	else {
+	if (o == NULL)
+		return -EINVAL;
 
-		if (non_fs_namespace) {
-			while (write(1, "", 0) < 0)
-				usleep(1000);
-			portCreate(&dummyfs_common.port);
-			if (portRegister(dummyfs_common.port, mountpt, &root) < 0) {
-				LOG("can't mount as %s\n", mountpt);
-				return -1;
-			}
-			mountpt = NULL;
-		}
-		else {
-			portCreate(&dummyfs_common.port);
-		}
+	if (!S_ISREG(o->mode))
+		ret = -EINVAL;
 
-	}
+	if (buff == NULL)
+		ret = -EINVAL;
 
-	if (mutexCreate(&dummyfs_common.mutex) != EOK) {
-		LOG("could not create mutex\n");
-		return 2;
+	if (ret != EOK) {
+		object_put(dummyfs, o);
+		return ret;
 	}
 
-	object_init();
-	dev_init();
+	object_lock(dummyfs, o);
 
-	/* Create root directory */
-	o = object_create();
+	ret = dummyfs_write_internal(dummyfs, o, offs, buff, len);
 
-	if (o == NULL)
-		return -1;
+	object_unlock(dummyfs, o);
+	object_put(dummyfs, o);
 
-	o->oid.port = dummyfs_common.port;
-	o->mode = S_IFDIR | 0666;
+	return ret;
+}
 
-	memcpy(&root, &o->oid, sizeof(oid_t));
-	dir_add(o, ".", S_IFDIR | 0666, &root);
-	dir_add(o, "..", S_IFDIR | 0666, &root);
 
-	if (!non_fs_namespace && mountpt == NULL) {
-		fetch_modules();
-		mountpt = remount_path;
-	}
+int dummyfs_createMapped(void *ctx, oid_t *dir, const char *name, void *addr, size_t size, oid_t *oid)
+{
+	dummyfs_t *dummyfs = (dummyfs_t *)ctx;
+	dummyfs_object_t *o;
+	dummyfs_chunk_t *chunk;
 
-	if (daemonize) {
-		/* mount synchronously */
-		if (!non_fs_namespace && dummyfs_mount_sync(mountpt)) {
-			LOG("failed to mount, exiting\n");
-			return 1;
-		}
+	if (dummyfs_create(ctx, dir, name, oid, 0755, otFile, NULL) != 0)
+		return -ENOMEM;
 
-		/* init completed - wake parent */
-		kill(getppid(), SIGUSR1);
+	if ((o = object_get(dummyfs, oid->id)) == NULL) {
+		dummyfs_destroy(ctx, oid);
+		return -EINVAL;
 	}
-	else if (mountpt != NULL) {
-		beginthread(dummyfs_mount_async, 4, &mtstack, sizeof(mtstack), (void*)mountpt);
+
+	if ((chunk = malloc(sizeof(dummyfs_chunk_t))) == NULL) {
+		object_put(dummyfs, o);
+		dummyfs_destroy(ctx, oid);
+		return -ENOMEM;
 	}
 
-	/*** MAIN LOOP ***/
+#ifdef MMU
+	addr = (void *)mmap(NULL, (size + 0xfff) & ~0xfff, 0x1 | 0x2, 0, OID_PHYSMEM, addr);
+	if (addr == MAP_FAILED) {
+		free(chunk);
+		object_put(dummyfs, o);
+		dummyfs_destroy(ctx, oid);
+		return -ENOMEM;
+	}
+#endif
 
-	LOG("initialized\n");
+	chunk->offs = 0;
+	chunk->size = size;
+	chunk->used = size;
+	chunk->data = (void *)((uintptr_t)addr & ~0xfff) + ((uintptr_t)addr & 0xfff);
+	chunk->next = chunk;
+	chunk->prev = chunk;
 
-	for (;;) {
-		if (msgRecv(dummyfs_common.port, &msg, &rid) < 0)
-			continue;
+	object_lock(dummyfs, o);
+	o->size = size;
+	o->mode = 0xaBadBabe;
```

Comment:
This constant is used in various places, so maybe let's make it defined in some header?